lib/hanami/commands/generate/model.rb

```diff
@@ -103,11 +103,24 @@ def repository_path
         # @since 0.9.1
         # @api private
         def migration_path
+          existing_migration_path || new_migration_path
+        end
+
+        # @since 1.1.0
+        # @api private
+        def new_migration_path
           timestamp = Time.now.utc.strftime(Migration::TIMESTAMP_FORMAT)
-          filename = Migration::FILENAME_PATTERN % { timestamp: timestamp, name: "create_#{table_name}"}
+          filename = Migration::FILENAME_PATTERN % { timestamp: timestamp, name: migration_name_underscored }
+
           Hanami::Model.configuration.migrations.join(filename)
         end
 
+        # @since 1.1.0
+        # @api private
+        def existing_migration_path
+          Utils::FileList["#{Hanami::Model.configuration.migrations}/[0-9]*_#{migration_name_underscored}.rb"].first
```

Comment:
FileList is a wrapper around `Dir.glob` which takes a pattern that is not a regex, but more like a shell glob. (See the [docs](https://ruby-doc.org/core-2.4.0/Dir.html#method-c-glob))

The way you wrote this pattern one is tempted to think that it would only match if a sequence of 0 or more numbers is followed by `_` and the underscored migration name `.rb`. However, it would in fact match anything that starts with one number and ends with the migration name. E.g., `20170420080151_revert_create_products.rb` would match too if the underscored migration name were `create_products`.